Core/ScriptLoader/Include/Babylon/ScriptLoader.h

```diff
@@ -23,7 +23,7 @@ namespace Babylon
 
         ~ScriptLoader();
 
-        void LoadScript(std::string url);
+        void LoadScript(std::string url, Napi::FunctionReference* onSuccess = nullptr, Napi::FunctionReference* onError = nullptr);
```

Comment:
This seems a little odd to me.
1. Having `Napi::FunctionReference`s as the callbacks doesn't seem right for a Core component which targets C++ consumers. Could these just be `std::function`s?
2. Is this call allowed to fail? I think that might not be allowed; and if not, can we just leave out the `onError` callback?
3. Also, if `LoadScript` isn't allowed to fail (or, more accurately, if it's not allowed to _survive_ failure), I think the same effective thing as an `onSuccess` callback could be achieved by giving `ScriptLoader` its own `Dispatch()` method, which would simply use the same mechanisms as are already within `ScriptLoader` to execute a given `function` linearly alongside all other `ScriptLoader` calls, similar to pre-existing patterns.